datafusion/expr/src/expr.rs

```diff
@@ -3929,6 +4056,366 @@ mod test {
         }
     }
 
+    #[test]
+    fn infer_placeholder_simple_case() {
```

Comment:
```suggestion
    fn infer_case_expr_placeholder_simple() {
```

The name was a bit confusing to me